src/test/java/org/javarosa/core/model/instance/geojson/GeoJsonExternalInstanceTest.java

```diff
@@ -0,0 +1,151 @@
+    @Test
+    public void parse_throwsException_ifTopLevelElementNotFeatureCollection() {
+        try {
+            GeoJsonExternalInstance.parse("id", new ByteArrayInputStream(GEOMETRY_COLLECTION_GEOJSON.getBytes(StandardCharsets.UTF_8)));
+            fail("Exception expected");
+        } catch (IOException e) {
+            // expected
+        }
+    }
+
+    @Test
+    public void parse_parsesMultipleFeatures() throws IOException {
+        TreeElement featureCollection = GeoJsonExternalInstance.parse("id", new ByteArrayInputStream(FEATURE_COLLECTION_GEOJSON.getBytes(StandardCharsets.UTF_8)));
+        assertThat(featureCollection.getNumChildren(), is(2));
+    }
+
+    @Test
+    public void parse_addsGeometryAsChild() throws IOException {
+        TreeElement featureCollection = GeoJsonExternalInstance.parse("id", new ByteArrayInputStream(FEATURE_COLLECTION_GEOJSON.getBytes(StandardCharsets.UTF_8)));
+        assertThat(featureCollection.getChildAt(0).getChild("geometry", 0).getValue().getValue(), is("102 0.5 0 0"));
+        assertThat(featureCollection.getChildAt(1).getChild("geometry", 0).getValue().getValue(), is("104 0.5 0 0"));
+    }
+
+    @Test
+    public void parse_addsAllOtherPropertiesAsChildren() throws IOException {
+        TreeElement featureCollection = GeoJsonExternalInstance.parse("id", new ByteArrayInputStream(FEATURE_COLLECTION_GEOJSON.getBytes(StandardCharsets.UTF_8)));
+        assertThat(featureCollection.getChildAt(0).getNumChildren(), is(4));
+        assertThat(featureCollection.getChildAt(0).getChild("name", 0).getValue().getValue(), is("My cool point"));
+
+        assertThat(featureCollection.getChildAt(1).getNumChildren(), is(5));
+        assertThat(featureCollection.getChildAt(1).getChild("special-property", 0).getValue().getValue(), is("special value"));
+    }
+
+    @Test
+    public void getOdkCoordinates_convertsPointGeoJsonGeometry() throws IOException {
+        String point = "{ \"type\": \"Point\", \"coordinates\": [ 102, 0.5 ] }";
+        ObjectMapper objectMapper = new ObjectMapper();
+        GeojsonGeometry geometry = objectMapper.readValue(point, GeojsonGeometry.class);
+        assertThat(geometry.getOdkCoordinates(), is("102 0.5 0 0"));
+    }
+
+    @Test
+    public void getOdkCoordinates_throwsException_ifGeometryNotPoint() throws IOException {
+        String notPoint = "{\n" +
+            "               \"type\": \"LineString\",\n" +
+            "               \"coordinates\": [102.0, 0.0]\n" +
+            "           }";
+        ObjectMapper objectMapper = new ObjectMapper();
+        GeojsonGeometry geometry = objectMapper.readValue(notPoint, GeojsonGeometry.class);
+
+        try {
+            geometry.getOdkCoordinates();
+            fail("Exception expected");
+        } catch (UnsupportedOperationException e) {
+            // expected
+        }
+    }
+
+    private static final String GEOMETRY_COLLECTION_GEOJSON = "{\n" +
```

Comment:
Pulled out a file since that addresses your comment about the `protected` method too. I used the constant because it was easier for me to jump to as I was writing the tests but I also see your point about the visual noise. Too bad we can't use Java 15 `"""`.

Reply:
:neckbeard: Kotlin has multiline strings with support all the way back to Java 6 (https://kotlinlang.org/docs/coding-conventions.html#strings).